<a href="https://colab.research.google.com/github/robertoarturomc/MetodosEstadisticos/blob/main/2_Reto_Modelado_Estadistico.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

### Reto Modelación Estadística
## 2. Funciones Avanzadas de Probabilidad

In [ ]:
df <- read.csv("Customer-Churn-Records.csv")
tail(df)

RowNumber CustomerId Surname   CreditScore Geography Gender Age Tenure
9995   9995     15719294   Wood      800         France    Female 29   2    
9996   9996     15606229   Obijiaku  771         France    Male   39   5    
9997   9997     15569892   Johnstone 516         France    Male   35  10    
9998   9998     15584532   Liu       709         France    Female 36   7    
9999   9999     15682355   Sabbatini 772         Germany   Male   42   3    
10000 10000     15628319   Walker    792         France    Female 28   4    
      Balance   NumOfProducts HasCrCard IsActiveMember EstimatedSalary Exited
9995       0.00 2             0         0              167773.55       0     
9996       0.00 2             1         0               96270.64       0     
9997   57369.61 1             1         1              101699.77       0     
9998       0.00 1             0         1               42085.58       1     
9999   75075.31 2             1         0               92888.52       1     
10000 130142.79 1             1         0               38190.78       0     
      Complain Satisfaction.Score Card.Type Point.Earned
9995  0        4                  PLATINUM  311         
9996  0        1                  DIAMOND   300         
9997  0        5                  PLATINUM  771         
9998  1        3                  SILVER    564         
9999  1        2                  GOLD      339         
10000 0        3                  DIAMOND   911

### Conteos
¿Cómo puedo saber cuántos y cuáles tipos distintos de *Card.Type* existen?

¿Cómo puedo saber cuántos y cuáles tipos distintos de *Satisfaction.Score* existen?

¿Cómo puedo saber cuántas y cuáles combinaciones de *Card.Type* con *Satisfaction.Score* existen?


In [ ]:
unique(df$Card.Type)

[1] "DIAMOND"  "GOLD"     "SILVER"   "PLATINUM"

In [ ]:
n_tarjetas <- length(unique(df$Card.Type))
n_tarjetas

[1] 4

In [ ]:
unique(df$Satisfaction.Score)

[1] 2 3 5 4 1

In [ ]:
n_scores <- length(unique(df$Satisfaction.Score))
n_scores

[1] 5

In [ ]:
n_tarjetas * n_scores

[1] 20

### Permutaciones

Para sacar las distintas permutaciones de *k* elementos en *n* elementos, teníamos la siguiente fórmula:

$permutaciones = \dfrac {n!}{(n-k)!}$

Podemos calcularlo directamente. Por ejemplo, para tener las permutaciones de 2 elementos en 5:

In [ ]:
factorial(5)/factorial(5-2)

[1] 20

Aunque también podemos imprimir las distintas permutaciones usando el paquete *gtools*.

In [ ]:
install.packages("gtools")

Installing package into ‘/usr/local/lib/R/site-library’
(as ‘lib’ is unspecified)



In [ ]:
library(gtools)

In [ ]:
permutations(5, 2)

[,1] [,2]
 [1,] 1    2   
 [2,] 1    3   
 [3,] 1    4   
 [4,] 1    5   
 [5,] 2    1   
 [6,] 2    3   
 [7,] 2    4   
 [8,] 2    5   
 [9,] 3    1   
[10,] 3    2   
[11,] 3    4   
[12,] 3    5   
[13,] 4    1   
[14,] 4    2   
[15,] 4    3   
[16,] 4    5   
[17,] 5    1   
[18,] 5    2   
[19,] 5    3   
[20,] 5    4

### Combinaciones

Para sacar las distintas combinaciones de *k* elementos en *n* elementos, teníamos la siguiente fórmula:

$combinaciones = \dfrac {n!}{k!(n-k)!} =  {n \choose k}$

Para tener las combinaciones de 2 elementos en 5:

In [ ]:
factorial(5)/(factorial(2) * factorial(5-2) )

[1] 10

Aunque también podemos imprimir las distintas combinaciones usando el paquete *gtools*.

In [ ]:
combinations(5, 2)

[,1] [,2]
 [1,] 1    2   
 [2,] 1    3   
 [3,] 1    4   
 [4,] 1    5   
 [5,] 2    3   
 [6,] 2    4   
 [7,] 2    5   
 [8,] 3    4   
 [9,] 3    5   
[10,] 4    5

Tanto *combinations* como *permutations* permiten ser más explícito, con algún vector en particular. Para el mismo ejemplo, si quisiéramos sacar todas las combinaciones y permutaciones tomando 2 elementos de las cinco vocales, podríamos realizarlo de la siguiente manera:

In [ ]:
permutations(5, 2, c("a", "e", "i", "o", "u"))

[,1] [,2]
 [1,] a    e   
 [2,] a    i   
 [3,] a    o   
 [4,] a    u   
 [5,] e    a   
 [6,] e    i   
 [7,] e    o   
 [8,] e    u   
 [9,] i    a   
[10,] i    e   
[11,] i    o   
[12,] i    u   
[13,] o    a   
[14,] o    e   
[15,] o    i   
[16,] o    u   
[17,] u    a   
[18,] u    e   
[19,] u    i   
[20,] u    o

In [ ]:
combinations(5, 2, c("a", "e", "i", "o", "u"))

[,1] [,2]
 [1,] a    e   
 [2,] a    i   
 [3,] a    o   
 [4,] a    u   
 [5,] e    i   
 [6,] e    o   
 [7,] e    u   
 [8,] i    o   
 [9,] i    u   
[10,] o    u

### Particiones

El número de particiones posibles de un conjunto de 𝑛 elementos
con 𝑟 particiones es:

$particiones = \dfrac {n!}{n_1!n_2! ... n_r!} =  {n!\choose n_1,n_2, ... n_r} $

Donde $𝑛_1$, $𝑛_2$, $𝑛_3$, ... , $𝑛_r$ son el número de elementos en cada una de las particiones

Por ejemplo, para dividir un grupo de 10 personas en parejas.


### Subconjuntos

El número de todos los subconjuntos posibles de un conjunto de 𝑛 objetos es igual a $2^n$

Por ejemplo, todos los subconjuntos posibles para un conjunto de 10 elementos es:


In [ ]:
factorial(10)/(5 * factorial(2))

[1] 362880

Ejemplos:

En una caja de pernos se encuentran 8 gruesos, 5
medianos y 3 angostos.
Una caja de tuercas contiene 6 que ajustan a
los pernos gruesos, 4 que ajustan a los pernos
medianos y 2 que ajustan a los pernos angostos.

Se elige aleatoriamente un perno y una tuerca
¿ Cuál es la probabilidad de que la tuerca ajuste
con el perno? R: 0.3854


In [ ]:
(8/16 * 6/12) + (5/16 * 4/12) + (3/16 * 2/12)

[1] 0.3854167

In [ ]:
df$NumOfProducts

[1] 1 1 3 2 1 2 2 4 2 1 2 2 2 2 2 2 1 2 1 2 2 2 1 2 1 2 1 2 1 2 3 1 2 2 2 1
   [37] 1 1 1 2 1 1 1 1 1 2 1 1 1 2 2 2 2 1 1 1 1 1 1 1 1 2 2 1 1 2 1 2 2 1 4 1
   [73] 1 2 2 2 2 1 1 2 2 1 2 1 2 2 1 2 3 2 3 2 2 2 2 1 1 2 2 2 1 2 1 2 1 2 1 1
  [109] 1 1 1 2 1 1 2 2 1 2 1 2 2 1 2 1 1 1 1 1 2 1 2 1 1 2 2 2 1 1 1 2 1 2 1 1
  [145] 1 1 2 1 1 2 2 1 2 1 2 1 2 2 1 1 1 2 1 2 1 2 1 1 2 2 1 2 1 1 2 1 2 2 1 1
  [181] 1 2 2 2 1 2 1 1 2 2 2 2 2 1 2 1 2 1 2 1 1 1 1 2 1 1 1 1 1 1 2 2 2 1 2 1
  [217] 1 2 2 1 2 1 2 1 1 2 2 3 1 1 2 2 2 2 2 1 1 3 3 2 2 1 2 1 2 1 2 2 1 1 1 2
  [253] 2 1 1 2 2 2 1 1 2 1 2 1 1 1 2 2 1 1 2 3 2 2 1 2 1 2 1 1 2 2 1 1 2 2 1 1
  [289] 1 2 1 2 2 1 2 1 1 3 1 2 2 1 2 2 1 1 1 2 1 1 2 1 2 1 1 2 1 2 1 1 1 1 1 2
  [325] 2 1 2 1 1 2 2 1 2 1 1 1 2 2 1 1 2 1 2 2 2 2 2 1 1 2 2 1 1 2 2 2 2 1 1 2
  [361] 2 1 1 2 2 1 2 2 2 1 1 2 2 1 2 1 3 1 1 3 2 2 2 2 1 1 2 2 2 2 1 1 2 2 1 2
  [397] 1 1 2 1 1 1 2 2 2 1 1 2 2 1 2 2 1 3 2 2 1 1 2 2 1 2 1 1 1 1 2 1 1 1 1 2
  [433] 1 1 1 1 2 2 2 1 1 2 2 1 1 1 1 1 1 2 1 2 1 1 1 2 1 2 1 1 2 2 2 1 2 1 2 3
  [469] 2 2 1 1 1 2 2 1 1 2 1 1 1 2 3 2 2 1 2 1 2 1 1 1 2 1 2 2 2 1 2 1 2 2 1 2
  [505] 1 2 1 1 2 2 2 2 2 1 1 2 3 2 2 1 1 1 2 1 1 2 2 2 1 1 2 1 1 1 1 1 2 2 3 1
  [541] 2 2 1 1 3 2 2 2 2 3 2 1 1 2 1 3 2 1 2 1 1 1 1 1 1 2 1 1 2 3 2 1 2 2 1 2
  [577] 1 2 1 2 2 2 2 1 2 1 1 1 1 2 1 1 1 1 2 1 2 1 1 1 3 2 1 1 1 1 2 2 2 1 1 2
  [613] 1 1 1 1 2 1 1 2 1 1 2 2 1 1 1 1 2 1 1 1 1 2 2 2 2 1 1 1 1 1 1 1 1 1 2 1
  [649] 2 2 2 1 1 1 2 1 1 2 1 2 1 1 2 2 2 2 1 2 1 1 2 2 1 2 2 2 2 1 2 2 1 1 1 2
  [685] 1 1 2 1 1 1 2 2 1 2 1 2 1 2 2 1 2 1 2 2 2 3 1 2 2 2 1 1 1 1 3 2 1 2 2 1
  [721] 2 2 2 1 1 2 2 3 1 1 1 2 1 2 1 1 1 1 2 2 1 1 1 3 2 2 1 1 1 2 2 1 1 2 2 2
  [757] 2 2 1 1 1 1 1 1 3 1 2 2 2 1 2 1 2 1 2 1 1 1 2 1 2 2 2 2 1 1 2 1 2 1 2 2
  [793] 1 1 1 2 1 1 2 1 2 2 2 2 2 1 1 2 1 1 2 2 2 3 2 2 2 1 1 2 2 1 1 1 2 2 2 2
  [829] 2 2 2 1 1 2 2 2 2 1 1 1 1 1 1 1 1 1 2 2 1 1 2 2 1 2 2 2 1 2 1 2 1 1 1 2
  [865] 2 2 2 1 2 1 1 3 2 2 2 1 1 2 1 2 1 3 1 1 1 2 1 2 1 1 1 2 1 3 2 1 2 1 1 1
  [901] 2 2 1 2 1 1 1 1 1 2 2 1 1 1 1 1 1 1 2 2 2 1 3 1 1 2 1 1 2 1 1 2 2 1 3 2
  [937] 1 1 1 1 1 2 1 1 1 2 1 1 1 1 2 1 1 1 2 1 2 2 1 1 1 1 1 3 1 1 1 2 1 2 1 2
  [973] 2 2 2 1 1 2 1 2 1 2 2 1 2 2 1 1 2 1 2 2 2 1 2 3 2 1 2 1 2 2 2 2 2 3 2 1
 [1009] 1 1 1 1 2 2 2 1 2 1 1 1 1 2 2 2 1 2 2 1 2 1 1 2 2 2 2 1 1 2 1 1 1 2 1 2
 [1045] 2 2 1 1 1 1 1 1 1 1 1 2 2 1 2 1 1 2 1 1 2 2 1 2 1 1 1 1 2 2 2 2 2 1 2 1
 [1081] 1 1 1 2 2 1 3 2 2 2 1 1 1 2 2 1 3 2 2 2 1 2 2 2 2 1 2 1 2 2 1 1 2 1 1 1
 [1117] 2 3 1 2 2 2 2 1 1 1 1 2 1 1 1 1 2 2 1 2 2 2 2 1 1 2 2 2 1 1 2 3 1 2 1 1
 [1153] 1 1 2 1 2 1 2 2 1 1 1 2 1 2 2 1 2 1 1 1 2 2 1 2 2 2 1 2 2 1 1 2 1 1 2 1
 [1189] 2 1 2 1 2 3 2 1 1 2 1 1 2 1 1 1 2 1 2 1 1 2 3 2 2 2 1 1 1 2 1 2 2 2 1 2
 [1225] 1 1 2 2 2 2 2 1 1 2 1 1 1 2 2 2 1 2 1 1 2 1 2 2 2 2 1 1 1 2 4 2 1 2 2 2
 [1261] 1 2 1 1 1 3 1 1 1 2 1 1 2 2 1 1 1 2 2 2 2 1 2 1 1 1 2 1 1 1 1 1 2 2 1 1
 [1297] 1 2 1 2 1 1 2 2 2 1 1 2 2 2 1 1 1 2 1 1 2 1 3 2 2 2 1 2 1 1 2 1 1 1 2 2
 [1333] 2 1 2 1 1 2 1 1 1 3 1 1 2 2 1 2 1 1 1 1 1 2 1 1 2 2 2 1 2 2 2 2 2 1 2 1
 [1369] 2 1 2 1 2 1 1 2 3 1 2 1 1 3 2 1 1 1 2 1 1 1 2 2 2 1 1 2 1 2 2 2 2 2 1 1
 [1405] 1 1 1 1 2 1 2 2 1 2 1 2 2 2 1 1 1 1 1 2 1 1 1 1 1 1 1 1 1 2 1 2 2 1 1 1
 [1441] 1 1 1 3 2 1 1 1 1 2 1 2 1 1 1 2 2 2 2 1 1 2 1 1 2 1 1 2 1 4 2 1 1 2 2 1
 [1477] 2 1 1 2 1 2 1 2 2 2 1 1 4 1 2 1 1 1 2 1 2 2 2 1 1 2 2 1 2 1 2 1 2 1 1 2
 [1513] 2 2 2 1 1 2 2 1 1 2 1 2 2 1 2 1 1 2 2 2 1 1 1 1 2 1 1 1 2 1 1 1 2 1 1 2
 [1549] 2 1 2 2 1 2 2 1 2 1 2 2 1 2 2 1 2 2 2 1 1 1 2 1 1 1 1 2 2 2 1 1 1 2 2 2
 [1585] 2 1 2 2 1 1 2 2 1 1 2 2 1 1 1 2 1 2 3 1 1 2 2 2 2 1 2 1 1 2 3 1 2 2 2 1
 [1621] 2 1 2 1 2 2 2 1 2 1 1 1 1 1 2 1 2 2 1 2 1 2 1 1 1 1 1 2 1 2 2 1 1 1 2 1
 [1657] 1 3 1 2 2 2 1 1 1 2 1 2 1 2 1 2 2 1 1 2 3 2 1 1 1 2 1 1 1 1 2 1 2 1 2 1
 [1693] 2 1 1 2 2 2 2 2 2 4 2 1 1 1 1 2 1 2 1 2 1 1 1 2 1 1 1 2 2 2 1 2 2 2 1 2
 [1729] 2 2 2 2 1 2 2 1 1 1 1 2 1 2 1 1 1 1 1 1 1 2 2 1 1 2 2 1 2 1 1 2 1 2 1 1
 [1765] 2 2 2 2 2 2 1 1 2 1 2 1 2 1 1 2 2 1 2 2 1 2 2 1 1 2 2 2 1 2 2 1 1 1 3 2
 [18

En mi tabla de clientes, tengo una variable, llamada *NumOfProducts*. Por privacidad, no puedo saber _qué_ productos tiene cada quién, pero sé que mis productos son:

In [ ]:
productos <- c("Crédito", "Seguro", "Hipoteca", "Inversion")

Para un cliente que tiene NumOfProducts=2, ¿en cuántas combinaciones posibles uno de sus productos es un Seguro?

In [ ]:
combinations(4,2, productos)

[,1]      [,2]     
[1,] Crédito   Hipoteca 
[2,] Crédito   Inversion
[3,] Crédito   Seguro   
[4,] Hipoteca  Inversion
[5,] Hipoteca  Seguro   
[6,] Inversion Seguro

Agrega una nueva columna que agregue todas el número combinaciones posibles de productos, sabiendo que hay 4 productos en total.


In [ ]:
df["Combinaciones_Productos"] <- factorial(4) / factorial(4 - df$NumOfProducts)

head(df)

RowNumber CustomerId Surname  CreditScore Geography Gender Age Tenure
1 1         15634602   Hargrave 619         France    Female 42  2     
2 2         15647311   Hill     608         Spain     Female 41  1     
3 3         15619304   Onio     502         France    Female 42  8     
4 4         15701354   Boni     699         France    Female 39  1     
5 5         15737888   Mitchell 850         Spain     Female 43  2     
6 6         15574012   Chu      645         Spain     Male   44  8     
  Balance   NumOfProducts HasCrCard IsActiveMember EstimatedSalary Exited
1      0.00 1             1         1              101348.88       1     
2  83807.86 1             0         1              112542.58       0     
3 159660.80 3             1         0              113931.57       1     
4      0.00 2             0         0               93826.63       0     
5 125510.82 1             1         1               79084.10       0     
6 113755.78 2             1         0              149756.71       1     
  Complain Satisfaction.Score Card.Type Point.Earned Combinaciones_Productos
1 1        2                  DIAMOND   464           4                     
2 1        3                  DIAMOND   456           4                     
3 1        3                  DIAMOND   377          24                     
4 0        5                  GOLD      350          12                     
5 0        5                  GOLD      425           4                     
6 1        5                  DIAMOND   484          12

## 2.b) Variables Aleatorias

In [ ]:
# unique() también sirve para encontrar mi espacio muestral

unique(df$Tenure)

[1]  2  1  8  7  4  6  3 10  5  9  0

Acá puedo ver que *Tenure* es una variable Discreta, con valores del 0 al 10.

Pero, ¿tendrá una distribución Uniforme? Veamos el conteo de cada uno de los posibles valores.

In [ ]:
table(df$Tenure)


   0    1    2    3    4    5    6    7    8    9   10 
 413 1035 1048 1009  989 1012  967 1028 1025  984  490 

¡Los valores varían mucho!
Mi Dataset es, finalmente, una muestra del Universo total de clientes. Por lo tanto, es altamente probable que la Variable Aleatoria *Tenure* **no** de distribuya Uniformemente.

Pero a ver: ¿puedo ir investigando cómo se distribuyen las variables en mi set de Datos? ¡Claro que sí!

In [ ]:
table(df$HasCrCard)


   0    1 
2945 7055 

In [ ]:
sum(df$HasCrCard) / dim(df)[1]

[1] 0.7055

Si tomo al azar un cliente, la probabilidad de que este tenga una tarjeta de Crédito se distribuirá **Bernoulli**, con p=0.7055

Recordatorio: una variable aleatoria de Bernoulli representa casos experimentales donde sólo existe éxito o fracaso (sucede o no sucede).

Si, en cambio, tomo $n$ clientes al azar, y puesto que cada uno de ellos son eventos independientes, la probabilidad de que $k$ de ellos tengan Tarjeta de Crédito se distribuye **Binomial**. Por ejemplo, si tomo una muestra de 5 elementos, la probabilidad de que tres de ellos tenga tarjeta de crédito es:

$p_x(k) = {n\choose k} p^k (1-p) ^{n-k} $


$p_x(3) = {5\choose 3} 0.7055^3 (1-0.7055) ^{5-3} $

In [ ]:
# Puedo crear una función que me ayude con el número de combinaciones.
n_comb <- function(n, k) {
  return(factorial(n) / (factorial(k) * factorial(n - k)))
}

n_comb(5, 3)

[1] 10

In [ ]:
# Entonces, la probabilidad sería:

p_3_en_5 <- n_comb(5, 3) * (0.7055)^3 * (1-0.7055)^2
p_3_en_5

[1] 0.3045521

Y puedo hacer un experimento para ver si es cierto. Aplicando la _Ley de los Grandes números_, sé que si repito suficientes veces el experimento, sabré que el valor irá "acercándose" a 0.30455 aproximadamente

In [ ]:
muestra <- sample(1:nrow(df), size=5,replace=FALSE)
muestra

[1] 7115 4445  769 6531 7429

Obtengamos 1000 muestras de 5 elementos, y veamos en cuántas de ellas hay 3 con Tarjeta de Crédito

In [ ]:
tarjetasCredito_muestras <- c()
for (i in 1:1000) {
muestra <- sample(1:nrow(df), size=5,replace=FALSE)
tarjetas_esta_muestra <- sum(df$HasCrCard[muestra])
tarjetasCredito_muestras <- c(tarjetasCredito_muestras, tarjetas_esta_muestra)
}

In [ ]:
tarjetasCredito_muestras

[1] 3 3 4 4 5 4 4 4 4 4 5 2 4 4 5 3 3 2 2 3 4 4 4 5 4 4 4 4 4 3 4 5 5 4 5 4 3
  [38] 4 5 4 3 4 5 4 4 4 5 4 4 2 1 5 4 3 4 5 4 4 3 4 3 5 3 3 3 4 4 4 4 4 4 2 3 3
  [75] 1 3 5 5 5 4 2 2 2 3 4 4 3 5 3 2 3 4 2 3 4 5 4 3 4 4 4 4 2 4 4 2 4 3 4 2 4
 [112] 5 3 3 5 5 4 2 3 3 4 2 4 4 3 4 4 5 3 4 5 4 3 4 4 3 5 4 3 3 3 1 4 4 3 5 3 4
 [149] 5 3 3 3 4 3 2 2 4 5 3 3 3 1 3 5 3 3 4 4 3 4 4 3 5 3 4 4 3 3 4 4 4 4 5 3 4
 [186] 4 5 3 4 4 3 2 3 3 3 4 2 4 3 3 3 4 3 4 1 5 4 4 5 4 4 3 1 3 4 4 4 4 3 4 4 4
 [223] 4 3 5 4 3 5 3 4 4 5 3 5 3 4 4 3 3 3 2 2 2 4 3 2 1 5 4 3 3 2 4 4 4 3 3 4 3
 [260] 3 5 3 3 3 1 2 5 3 5 3 4 4 3 3 3 3 2 4 3 3 5 4 3 3 5 3 0 3 2 3 2 4 3 2 2 4
 [297] 2 3 3 4 3 1 3 5 2 3 4 2 0 3 2 5 4 4 5 2 4 4 4 5 4 3 4 2 4 4 3 3 3 4 3 4 5
 [334] 4 4 5 5 5 4 3 2 3 3 1 1 4 4 3 5 3 1 1 4 5 5 2 3 5 2 4 4 1 4 3 3 3 3 2 4 4
 [371] 5 3 5 2 5 4 3 5 1 2 2 4 5 4 2 4 5 3 2 5 5 3 3 5 4 5 4 5 2 3 5 2 5 3 3 3 3
 [408] 4 5 5 5 4 4 4 4 3 5 4 5 5 3 3 3 5 4 3 4 1 5 3 4 3 5 4 4 4 5 4 3 3 4 3 3 4
 [445] 4 4 5 3 2 5 5 5 3 5 2 4 5 4 5 4 4 3 3 3 4 2 4 5 5 1 3 3 4 4 3 4 4 2 3 3 3
 [482] 4 5 4 4 4 4 5 5 2 4 4 4 4 4 4 4 1 4 3 4 5 4 3 3 4 4 3 4 4 2 3 5 2 5 4 5 3
 [519] 3 4 4 5 4 4 5 3 4 4 4 5 3 4 4 4 2 4 2 2 4 3 3 2 5 4 3 4 4 5 5 3 4 0 2 4 4
 [556] 4 3 4 3 3 3 5 4 2 4 2 3 3 5 4 4 3 4 3 3 3 2 3 4 4 5 4 4 3 3 4 4 1 3 4 2 4
 [593] 3 5 4 4 4 3 4 4 3 2 4 4 3 4 5 3 2 2 2 3 5 2 4 3 3 3 5 3 5 3 3 4 5 5 5 4 3
 [630] 4 2 3 5 4 4 4 4 3 4 5 3 5 5 4 5 2 3 3 4 4 3 5 4 3 3 2 4 2 4 5 3 1 4 4 2 4
 [667] 1 2 3 5 3 4 4 3 2 3 5 4 4 4 5 4 5 4 3 3 3 4 3 4 4 5 3 4 3 3 5 3 5 4 3 5 3
 [704] 5 5 4 3 5 4 3 3 5 3 2 2 4 2 4 3 4 2 4 5 3 4 3 5 4 4 5 2 2 4 4 4 3 3 3 4 4
 [741] 4 4 3 3 5 5 2 3 3 4 4 3 4 2 4 5 5 3 5 4 3 4 5 3 3 2 3 4 2 3 4 2 4 3 4 3 3
 [778] 3 3 4 4 5 3 4 4 2 2 2 3 4 1 3 5 4 3 3 4 2 4 3 3 3 4 4 4 4 5 5 5 4 4 3 4 3
 [815] 4 2 2 4 5 3 4 2 4 3 5 4 2 4 5 4 5 3 4 5 3 5 4 4 2 2 5 4 3 4 4 3 4 4 2 4 3
 [852] 5 2 5 3 4 3 2 4 3 3 3 2 3 2 4 3 1 4 1 2 5 4 4 5 3 5 3 3 4 5 5 5 5 2 5 5 3
 [889] 4 5 3 4 4 5 1 4 4 3 3 4 1 4 3 3 2 4 2 5 2 3 1 4 4 3 4 1 4 4 3 4 3 3 4 2 2
 [926] 3 2 5 4 5 5 4 4 5 3 4 5 4 4 5 5 1 2 2 4 3 3 4 4 5 5 5 2 4 4 3 4 1 3 3 5 4
 [963] 4 4 5 3 3 3 4 3 5 2 5 4 3 5 4 2 3 4 3 5 2 2 4 2 5 4 4 4 4 3 3 3 3 5 4 5 5
[1000] 3

In [ ]:
table(tarjetasCredito_muestras)

tarjetasCredito_muestras
  0   1   2   3   4   5 
  3  30 117 296 369 185 

In [ ]:
30 / 1000

3 
0.296

Valor que es, efectivamente, aproximado, al 30.45 % que había calculado.

De la misma manera, sería interesante analizar de qué manera se distribuirían las demás variables. Por ejemplo, *Age* y *Tenure*, al hacer referencia al tiempo, quizá se distribuyan Poisson...

### Valor esperado de una Variable Discreta

Lo definimos como el posible valor de la variable, por la probabilidad de que tome ese valor. Es decir, un promedio ponderado.

$E(X) ) \sum x p_x(X) $

In [ ]:
# ¿Cuál será el valor Esperado de Tenure?

table(df$Tenure)


   0    1    2    3    4    5    6    7    8    9   10 
 413 1035 1048 1009  989 1012  967 1028 1025  984  490 

In [ ]:
# Calculo las probabilidades individuales
table(df$Tenure) / 10000


     0      1      2      3      4      5      6      7      8      9     10 
0.0413 0.1035 0.1048 0.1009 0.0989 0.1012 0.0967 0.1028 0.1025 0.0984 0.0490 

In [ ]:
# Calculo el valor esperado
esperado_Tenure <- sum( (table(df$Tenure) / 10000) * c(0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10) )
esperado_Tenure

[1] 5.0128

In [ ]:
# Lo cuál, como sabemos, sería igual al promedio:
mean(df$Tenure)

[1] 5.0128

### Varianza y Desviación Estándar

In [ ]:
# Puedo calcular la Varianza de una variable de la siguiente manera:
var(df$Tenure)

[1] 8.364673

In [ ]:
# Y la desviación estándar de dicha variable, es la raíz cuadrada de la varianza:
sqrt(var(df$Tenure))

[1] 2.892174

In [ ]:
sd(df$Tenure)

[1] 2.892174